blog/2021/10/julia-1.7-highlights.md

```diff
@@ -0,0 +1,539 @@
+
+Here is how the syntax is lowered:
+```julia
+[a ;;; b ;;; c] == hvncat(3, a, b, c) # row_first isn't needed here
+[a b ;;; c d] == hvncat((1, 2, 2), true, a, b, c, d)
+[a b ; c ;;; d ;;;;] == hvncat(((2, 1, 1), (3, 1), (4,), (4,)), true, a, b, c, d)
+ #=
+ How the shape argument is constructed:
+ ___   _     _
+ 2     1     1 = (2, 1, 1) elements in each row
+ _______     _
+ 3           1 = (3, 1) elements in each column
+ _____________
+ 4             = (4,) elements in each 3d slice
+ _____________
+ 4             = (4,) elements in each 4d slice
+ =#
+```
+
+Last, but certainly not least, this syntax makes it very easy to write a one-column matrix or a 2+ dimension single-element array, a highly requested feature:
+```julia-repl
+julia> [1; 2; 3;;]
+3×1 Matrix{Int64}:
+ 1
+ 2
+ 3
+
+julia> [1;;;]
+1×1×1 Array{Int64, 3}:
+[:, :, 1] =
+ 1
+```
+
+For fun, here's an example of how you could use this syntax to generate a Sudoku-like board (h/t Matt Bauman):
+```julia
+[rand(1:9, 3, 3);;; rand(1:9, 3, 3);;; rand(1:9, 3, 3);;;;
+ rand(1:9, 3, 3);;; rand(1:9, 3, 3);;; rand(1:9, 3, 3);;;;
+ rand(1:9, 3, 3);;; rand(1:9, 3, 3);;; rand(1:9, 3, 3);;;;]
+```
+
+## Property Destructuring
+
+*Simeon Schaub*
+
+One fairly simple but hopefully useful new syntax addition is support for destructuring objects not only by iteration, but also by property name. This is modeled after the `(; a b)` syntax added in 1.5 for constructing `NamedTuple`s whose field names correspond to the used variable names.
+
+A simple example of how this syntax can be used:
+
+```julia-repl
+julia> nt = (a=1, b=2)
+(a = 1, b = 2)
+
+julia> (; a, b) = nt
+(a = 1, b = 2)
+
+julia> a
+1
+
+julia> b
+2
+```
+
+`(; a, b) = nt` is simply equivalent to `a = nt.a; b = nt.b`, so it can be used not just for `NamedTuple`s, but for destructuring the properties of any object and will also work with `DataFrame`s for example. It can also be used inside function arguments just like regular destructuring. A `myreal` function which just extracts the real part of a complex number could now be written as follows:
+
+```julia-repl
+julia> myreal((; re)::Complex) = re
+myreal (generic function with 1 method)
+
+julia> myreal(2 + 3im)
+2
+```
+
+This syntax is also compatible with type annotations (note that these don't play any role in dispatch when used inside function arguments though):
+
+```julia-repl
+julia> let
+           (; a::UInt, b::Float64) = (a=1, b=2)
+           b = 5
+           a, b
+       end
+(0x0000000000000001, 5.0)
+```
+
+## Conclusion
+
+Please enjoy the release, and as always [let us know](https://github.com/JuliaLang/julia/issues) if you encounter any problems or have any suggestions. We hope to be back in about four months to report on even more progress in version 1.8!
```

Comment:
Is four months a bit too optimistic? :sweat_smile:

Reply:
Hehe, well, we do hope for four months, don't we? :P

Reply:
Maybe "a few months"? :smile: